core/domain/question_services.py

```diff
@@ -483,7 +507,7 @@ def replace_skill_id_for_all_questions(
     question_models.QuestionModel.put_multi_questions(new_questions)
 
 
-def get_question_summaries_and_skill_descriptions(
+def get_displayable_question_skill_link_details(
```

Comment:
I feel like this function is returning way too much and is kind of confusing. Can we simplify this? Few ways I can think of are - 
1. Keep two separate fetch functions : one for QuestionSummary, one for skill summaries + skill difficulties (but this might be slower depending on how we fetch it from backend)
Alternatively: 
2. Instead of sending a list(list(str)) and a list(list(int)) let's combine these two to make list(list(SkillSummaryAndDifficultyObject)). This won't change how we fetch from the datastore but just how we structure the data back to the controller

Reply:
Implemented the second method. Found an already existing object `QuestionSkillLink`, that had all the required  fields but for a single object, so created a `MergedQuestionSkillLink` object in question_domain for this purpose.

Reply:
That's awesome thanks!